modules/core/src/main/java/org/apache/ignite/internal/processors/service/ServiceDeploymentActions.java

```diff
@@ -43,11 +48,42 @@
     /** Services deployment errors. */
     private Map<IgniteUuid, Collection<byte[]>> depErrors;
 
+    /** Kernal context. */
+    private final GridKernalContext ctx;
+
+    /**
+     * @param ctx Kernal context.
+     */
+    public ServiceDeploymentActions(GridKernalContext ctx) {
+        this.ctx = ctx;
+    }
+
     /**
      * @param servicesToDeploy Services info to deploy.
      */
     public void servicesToDeploy(@NotNull Map<IgniteUuid, ServiceInfo> servicesToDeploy) {
-        this.servicesToDeploy = servicesToDeploy;
+        Map<IgniteUuid, ServiceInfo> res = new HashMap<>();
+        for (Map.Entry<IgniteUuid, ServiceInfo> kv: servicesToDeploy.entrySet()) {
+            ServiceInfo dsc = kv.getValue();
+
+            ServiceConfiguration cfg = dsc.configuration();
+
+            if (!ctx.platform().hasContext() ||
+                !PlatformUtils.PLATFORM_DOTNET.equals(ctx.platform().context().platform())) {
+                if (cfg instanceof LazyServiceConfiguration) {
+                    String svcClsName = ((LazyServiceConfiguration)cfg).serviceClassName();
+
+                    if (PlatformDotNetServiceImpl.class.getName().equals(svcClsName))
+                        continue;
+                }
+                else if (cfg instanceof ServiceConfiguration && cfg.getService() instanceof PlatformDotNetServiceImpl)
+                    continue;
+            }
```

Comment:
This set of conditions is quite hard to understand. Can we do something like this?

```
var nodePlatform = ctx.platform().hasContext() ? ctx.platform().context() : "";
var servicePlatform = getServicePlatform(cfg); // Extract a method to detect platform name by service class

if (!nodePlatform.equals(servicePlatform))
    continue;
```

Reply:
Fixed.